### ISSUE TITLE
Nano with Axios upgrade does not work for HTTPS based couchdb access anymore

### ISSUE THREAD

**Issue by indrajit-kanjilal** [apache/couchdb-nano#271]
Nano has stopped working after moving to Axios for Secure Socket connections (https).
The following code which worked in nano@8.2.2 , does not work in current version

_--------------------------------------------------------------------------------------------------------
const Path = require('path');
const fs = require('fs');
const caFile = fs.readFileSync(`${Path.join(__dirname, 'secrets')}${Path.sep}cacert.crt`);

const dboptions = {
    url: 'https://XXXX:YYYYYY@localhost:6984',
    requestDefaults: { 
        ca : caFile
        //rejectUnauthorized: false
    }
  }
const nano = require('nano')(dboptions);
//const solwbusagedb = nano.db.use('solwbusage');

const start = async () => {
    try{
        await nano.db.list();
    }
    catch(err){
        console.log(err);
    }
}

start();
---------------------------------------------------------------------------------------------------_

## Expected Behavior

If upgrade to based versions of nano needs special handling of clients
which **requires** HTTPS based access to nano, it sahould be documented.

Either the code mentioned above should work as-is , or there should be a way to
make it work, which is documented.

## Current Behavior

nano@8.2.2 : **Works fine**

nano@latest:
Error: error happened in your connection
    at responseHandler (C:\..........\server\node_modules\nano\lib\nano.js:120:16)
    at C:\...........\server\node_modules\nano\lib\nano.js:405:13
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

## Steps to Reproduce (for bugs)

1. Just run the above code on a HTTPS enabled server on port 6984 with the CA certificate in "secrets" directory
2. Run it with nano@8.2.2 installed first , and then with latest version
3.
4.

## Context

I use it in one of our internal tool, and all our server policy mandates HTTPS enables ports only

## Your Environment

* Version used: Current (9*) and 8.2.2
* Browser Name and version:
* Operating System and version (desktop or mobile): Windows 10, node 12.16.1
* Link to your project: N/A

**Comment by ceras-ella** [comment]
I second this issue... Is anyone looking into it by the way?

And to add more details to it:
@indrajit-kanjilal you can list the databases using the following requestDefaults:
```
requestDefaults = {
  agent: new https.Agent({
    ca : caFile
    ...
    rejectUnauthorized: true
  })
};
```

which is all fine, but when you'll try to `create` or `destroy` a database you end up with a circular structure error that I believed it's been addressed in the following PR:
https://github.com/apache/couchdb-nano/pull/270 

so, any chance we can get some traction on this? we'll be reverting to version 8 until this is properly fixed.

**Comment by glynnbird** [comment]
Fixed in 9.0.4
